fitbenchmarking/parsing/fitbenchmark_parser.py

```diff
@@ -309,6 +312,48 @@ def _get_starting_values(self):
 
         return starting_values
 
+    def _get_mantid_starting_values(self):
+        """
+        Get the starting values for the problem
+
+        :return: Starting values for the function
+        :rtype: list of OrderedDict
+        """
+        # Mantid functions can have reserved keywords so ignore these
+        ignore = ['name', 'ties', 'constraints']
+
+        name_template = '{1}' if len(self._parsed_func) == 1 else 'f{0}_{1}'
+        starting_values = []
+        for i, f in enumerate(self._parsed_func):
+
+            # use mantid function factory to seperate attributes and parameters
+            fcopy = f.copy()
+            if 'ties' in fcopy:
+                fcopy.pop('ties')
+            fstring = ", ".join(f"{key}={value}"
+                                for key, value in fcopy.items())
+            ffun = msapi.FunctionFactory.createInitialized(fstring)
+            attr_names = [s for s in ffun.attributeNames()]
+
+            # filter parameters
+            params = f.copy()
+            for key in ignore:  # filter reserved keywords from params
+                if key in params:
+                    params.pop(key)
+            for attr in attr_names:  # filter attributes from params
+                if attr in params:
+                    params.pop(attr)
+            if 'ties' in f:  # filter tied params from params
+                for tie in f['ties']:
+                    if tie in params:
+                        params.pop(tie)
```

Comment:
I'd be tempted to invert this and only take the params that are required using something like:
```
params = [(f.getParamName(i), f.getParamValue(i))
          for i in range(f.nParams())
          if f.isFixed(i)]
```

Although maybe that doesn't handle the more complicated ties properly.

Reply:
Agreed, that should work in exactly the same way with much less code!

Reply:
The reason it's done this way is that there are quite a few 'hidden' parameters in a lot of Mantid functions that the user has no interest in, some are even set to internal mantid vectors so best not to expose these.

Reply:
Ahh in that case this is fine!

Reply:
I take it back, I misunderstood how Mantid does things, your suggestion is the correct approach here as all non-fixed parameters that are optimised over should be exposed, will add this in.